src/nvim/path.c

```diff
@@ -373,10 +373,6 @@ void add_pathsep(char *p)
 char *FullName_save(char *fname, bool force)
   FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
 {
-  if (fname == NULL) {
-    return NULL;
-  }
```

Comment:
That was my concern. I apologize. I meant to raise this concern for
discussion and got pulled away right after submitting.
On May 7, 2015 10:31 PM, "Scott Prager" <notifications@github.com> wrote:

> In src/nvim/path.c
> <https://github.com/neovim/neovim/pull/2610#discussion_r29915053>:
>
> > @@ -373,10 +373,6 @@ void add_pathsep(char *p)
> >  char *FullName_save(char *fname, bool force)
> >    FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
> >  {
> > -  if (fname == NULL) {
> > -    return NULL;
> > -  }
>
> Sorry if I advised this, but unless we can prove that FullName_save()
> never gets called with a NULL pointer, removing this isn't safe and the
> FUNC_ATTR_NONNULL_RET must be removed to prevent invalid optimizations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neovim/neovim/pull/2610/files#r29915053>.
>